### ISSUE TITLE
Severe performance degradation with 1.3.0

### ISSUE THREAD

**Issue by emilianavt** [microsoft/onnxruntime#4421]
**Describe the bug**
More than 3-5 times longer inference times on models with onnxruntime 1.3.0 than with 1.2.0 and 1.1.2. The models were converted from PyTorch models based on the [geffnet](https://github.com/rwightman/gen-efficientnet-pytorch/tree/master/geffnet) implementation of MobileNetV3. They were exported using opset 11, as they contain bilinear upsample layers with `align_corners=True`.

The models are loaded using version 1.3.0 of onnxruntime with `GraphOptimizationLevel.ORT_ENABLE_ALL`. The issue both occurs with models directly after exporting them from PyTorch and models that have been preoptimized with 1.2.0 using `ORT_ENABLE_EXTENDED`, but it appears to be worse for the preoptimized models.

The original PyTorch model definitions are given [here](https://github.com/emilianavt/OpenSeeFace/blob/master/model.py).

**Urgency**
If there are particular important use cases blocked by this or strict project-related timelines, please share more information and dates. If there are no hard deadlines, please specify none.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux
- ONNX Runtime installed from (source or binary): pip
- ONNX Runtime version: 1.3.0
- Python version: 3.7.3
- Visual Studio version (if applicable): -
- GCC/Compiler version (if compiling from source): -
- CUDA/cuDNN version: running on CPU
- GPU model and memory: running on CPU (AMD Ryzen Threadripper 2950X)

**To Reproduce**
Code to reproduce the issue:

    from pathlib import Path
    import numpy as np
    import time
    import torch
    import torch.nn as nn
    import onnxruntime

    models = {"gaze.onnx": (2,3,32,32),
              "gaze_opt.onnx": (2,3,32,32),
              "detection.onnx": (1,3,224,224),
              "detection_opt.onnx": (1,3,224,224),
              "lm_model0.onnx": (1,3,224,224),
              "lm_model0_opt.onnx": (1,3,224,224),
              "lm_model1.onnx": (1,3,224,224),
              "lm_model1_opt.onnx": (1,3,224,224),
              "lm_model2.onnx": (1,3,224,224),
              "lm_model2_opt.onnx": (1,3,224,224),
              "lm_model3.onnx": (1,3,224,224),
              "lm_model3_opt.onnx": (1,3,224,224)}

    print(f"Versions:\n- pytorch: {torch.__version__}\n- numpy: {np.__version__}\n- onnxruntime: {onnxruntime.__version__}")
    print("============================")

    def benchmark(filename, size, runs=10000):
        print(f"Benchmarking model models/{filename} over {runs} runs.")
        np.random.seed(42)
        x = np.float32(np.random.random_sample(size) - 0.5)

        options = onnxruntime.SessionOptions()
        options.inter_op_num_threads = 1
        options.intra_op_num_threads = 4
        options.execution_mode = onnxruntime.ExecutionMode.ORT_SEQUENTIAL
        options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_ENABLE_ALL
        session = onnxruntime.InferenceSession(f"models/{filename}", sess_options=options)
        session.run([], {'input': x})

        total_time = 0.
        for i in range(runs):
            start = time.perf_counter()
            session.run([], {'input': x})
            total_time += time.perf_counter() - start
        avg_time = 1000. * (total_time / runs)

        print(f"Average duration: {avg_time:.2f}ms")

    for model in models.keys():
        benchmark(model, models[model])

The model files archive is too big to attach it here, so I put it on my Google drive.

https://drive.google.com/file/d/12Bwg-_dK5Yh48oU-zRURZxwcm3eaOryE/view?usp=sharing

**Expected behavior**
Inference time with 1.3.0 should at least be in line with 1.2.0.

**Benchmark results**
Version 1.1.2:

    Versions:
    - pytorch: 1.4.0
    - numpy: 1.18.2
    - onnxruntime: 1.1.2
    ============================
    Benchmarking model models/gaze.onnx over 10000 runs.
    Average duration: 1.39ms
    Benchmarking model models/gaze_opt.onnx over 10000 runs.
    Average duration: 1.33ms
    Benchmarking model models/detection.onnx over 10000 runs.
    Average duration: 8.01ms
    Benchmarking model models/detection_opt.onnx over 10000 runs.
    Average duration: 8.02ms
    Benchmarking model models/lm_model0.onnx over 10000 runs.
    Average duration: 6.40ms
    Benchmarking model models/lm_model0_opt.onnx over 10000 runs.
    Average duration: 6.20ms
    Benchmarking model models/lm_model1.onnx over 10000 runs.
    Average duration: 8.24ms
    Benchmarking model models/lm_model1_opt.onnx over 10000 runs.
    Average duration: 8.37ms
    Benchmarking model models/lm_model2.onnx over 10000 runs.
    Average duration: 11.81ms
    Benchmarking model models/lm_model2_opt.onnx over 10000 runs.
    Average duration: 11.75ms
    Benchmarking model models/lm_model3.onnx over 10000 runs.
    Average duration: 13.57ms
    Benchmarking model models/lm_model3_opt.onnx over 10000 runs.
    Average duration: 13.70ms

Version 1.2.0:

    Versions:
    - pytorch: 1.4.0
    - numpy: 1.18.2
    - onnxruntime: 1.1.2
    ============================
    Benchmarking model models/gaze.onnx over 10000 runs.
    Average duration: 1.39ms
    Benchmarking model models/gaze_opt.onnx over 10000 runs.
    Average duration: 1.33ms
    Benchmarking model models/detection.onnx over 10000 runs.
    Average duration: 8.01ms
    Benchmarking model models/detection_opt.onnx over 10000 runs.
    Average duration: 8.02ms
    Benchmarking model models/lm_model0.onnx over 10000 runs.
    Average duration: 6.40ms
    Benchmarking model models/lm_model0_opt.onnx over 10000 runs.
    Average duration: 6.20ms
    Benchmarking model models/lm_model1.onnx over 10000 runs.
    Average duration: 8.24ms
    Benchmarking model models/lm_model1_opt.onnx over 10000 runs.
    Average duration: 8.37ms
    Benchmarking model models/lm_model2.onnx over 10000 runs.
    Average duration: 11.81ms
    Benchmarking model models/lm_model2_opt.onnx over 10000 runs.
    Average duration: 11.75ms
    Benchmarking model models/lm_model3.onnx over 10000 runs.
    Average duration: 13.57ms
    Benchmarking model models/lm_model3_opt.onnx over 10000 runs.
    Average duration: 13.70ms

Version 1.3.0 (For this version the variance is also quite high. I get quite different results upon running the benchmark again, but still very slow.):

    Versions:
    - pytorch: 1.4.0
    - numpy: 1.18.2
    - onnxruntime: 1.3.0
    ============================
    Benchmarking model models/gaze.onnx over 10000 runs.
    Average duration: 6.09ms
    Benchmarking model models/gaze_opt.onnx over 10000 runs.
    Average duration: 6.41ms
    Benchmarking model models/detection.onnx over 10000 runs.
    Average duration: 30.44ms
    Benchmarking model models/detection_opt.onnx over 10000 runs.
    Average duration: 46.08ms
    Benchmarking model models/lm_model0.onnx over 10000 runs.
    Average duration: 25.53ms
    Benchmarking model models/lm_model0_opt.onnx over 10000 runs.
    Average duration: 28.35ms
    Benchmarking model models/lm_model1.onnx over 10000 runs.
    Average duration: 33.52ms
    Benchmarking model models/lm_model1_opt.onnx over 10000 runs.
    Average duration: 42.98ms
    Benchmarking model models/lm_model2.onnx over 10000 runs.
    Average duration: 45.63ms
    Benchmarking model models/lm_model2_opt.onnx over 10000 runs.
    Average duration: 43.23ms
    Benchmarking model models/lm_model3.onnx over 10000 runs.
    Average duration: 42.84ms
    Benchmarking model models/lm_model3_opt.onnx over 10000 runs.
    Average duration: 42.48ms

**Comment by askhade** [comment]
@emilianavt : For version 1.3 and above for the packages built with openmp instead of setting intra_op_num_threads you should set the environment variable OMP_NUM_THREADS. 

For more details you can check https://github.com/microsoft/onnxruntime/blob/master/docs/FAQ.md#how-do-i-force-single-threaded-execution-mode-in-ort-by-default-sessionrun-uses-all-the-computers-cores

Can you test this again and let me know if you are still seeing perf degradation?

**Comment by emilianavt** [comment]
Thank you. Setting `OMP_NUM_THREADS` solves this issue. However, controlling the number of intra threads globally introduces the issue that it is no longer possible to run multiple sessions with different numbers of intra threads in parallel.
